graphql-dgs-spring-graphql/src/main/kotlin/com/netflix/graphql/dgs/springgraphql/DgsGraphQLSourceBuilder.kt

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright 2024 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.graphql.dgs.springgraphql
+
+import com.netflix.graphql.dgs.internal.DefaultDgsQueryExecutor
+import com.netflix.graphql.dgs.internal.DgsSchemaProvider
+import graphql.GraphQL
+import graphql.execution.instrumentation.ChainedInstrumentation
+import graphql.execution.instrumentation.Instrumentation
+import graphql.schema.*
+import graphql.schema.GraphQLSchema.BuilderWithoutTypes
+import graphql.schema.idl.RuntimeWiring
+import graphql.schema.idl.TypeDefinitionRegistry
+import org.springframework.core.io.Resource
+import org.springframework.graphql.execution.*
+import org.springframework.graphql.execution.GraphQlSource.SchemaResourceBuilder
+import org.springframework.lang.Nullable
+import java.util.*
+import java.util.function.BiFunction
+import java.util.function.Consumer
+
+class DgsGraphQLSourceBuilder(private val dgsSchemaProvider: DgsSchemaProvider, private val reloadSchemaIndicator: DefaultDgsQueryExecutor.ReloadSchemaIndicator) : SchemaResourceBuilder {
```

Comment:
The `GraphQlSource.Builder` hierarchy has 2 alternatives. `GraphQlSource.Builder` itself exposes options independent of how the `GraphQLSchema` is created. `SchemaResourceBuilder` on the other hand is for builders that create the `GraphQLSchema` by parsing schema files, and building a `RuntimeWiring`. On the implementation side, `AbstractGraphQlSourceBuilder` has the common parts, while `DefaultSchemaResourceGraphQlSourceBuilder` implements `SchemaResouceBuilder` and prepares `GraphQLSchema` from those inputs, while 

This was originally created for SDL vs programmatic API in GraphQL Java, so if anyone wants to build `GraphQLSchema` programmatically they can extend `AbstractGraphQlSourceBuilder`. The same applies to DGS as well I think, and even though it also parses schema files and does things the SDL way, it has its own way of building the `GraphQLSchema` so it makes to extend `AbstractGraphQlSourceBuilder` and override the `initGraphQlSchema` method to call `dgsSchemaProvider.schema()`. So that will become a very trivial class and it won't have to repeat anything from `AbstractGraphQlSourceBuilder` as it currently does.

Likewise, I don't think `DgsGraphQLSourceBuilder` should have to be an implementation of `SchemaResourceBuilder`, but rather it only needs to be an `AbstractGraphQlSourceBuilder` which implements the base `GraphQlSource.Builder.` As it is currently, the builder doesn't have anything meaningful it can do with the extra options it gets from implementing the `SchemaResourceBuilder` subtype.

Reply:
Thanks for the comments. What you suggest makes sense to us. Looking at the latest sources, AbstractGraphQlSourceBuilder is not public. Could you update that so we can use it?

Reply:
You're right. I think we started out with just one package private builder before the hierarchy evolved. It does make sense for the base class to be public as it's more broadly useful, and I've made that change https://github.com/spring-projects/spring-graphql/commit/5fcf803680494f642b04f864c3179a34fb45ab25.